lib/linked_in/api.rb

```diff
@@ -124,5 +128,34 @@ def no_access_token_error
       msg = LinkedIn::ErrorMessages.no_access_token
       LinkedIn::InvalidRequest.new(msg)
     end
+
+    def verify_api_version!(version)
+      @api_version = if version.nil?
+        recent_api_version
+      else
+        validate_api_version!(version)
+      end
+    end
+
+    ## Verify the LinkedIn API version provided from caller.
+    # LinkedIn Marketing API Program will publish new versions monthly,
+    # and those versions will be supported for a minimum of one year.
+    # Raise error if version is older than year.
+    def validate_api_version!(version)
+      date = Date.strptime(version, '%Y%m')
+      if (Date.today - 365) > date
+        msg = LinkedIn::ErrorMessages.unsupported_api_version
+        raise LinkedIn::InvalidRequest.new(msg)
+      end
```

Comment:
Not really needed but I feel it is fine to show warning before a month of LinkedIn API version expiry.
```suggestion
      else (Date.today - 335) >= date
        warn('WARNING: Use the most recent LinkedIn API version to prevent interruptions since your current version of the LinkedIn API is set to expire after a month.')
      end
```